Add Query OGC API with CQL Filters sample to sample viewers

Gela opened the pull request:
This PR adds a sample demonstrating how to query and display features from an OgcFeatureCollectionTable to the sample viewers.

This sample showcases how to query an OGC Feature Collection Table using CQL. It includes a UI that provides the user with a drop down menu containing pre-populated CQL filters, a text field to enter max features, and two optional date pickers for start and end points so the results can be filtered by date range.

tanneryould reviewed (changes requested):
Everything looks great! I wish I could give more feedback, but this code looks really clean and well written. The one thing is I think you need to remove `.../QueryOGCAPICQLFilters/QueryOGCAPICQLFilters.png` from both directories. Unless I'm mistaken, we only need the screenshot.png and those are the same files.

Gela commented:
Thank you so much @tanneryould . I removed the duplicate png files and addressed a couple of other minor issues, please take a look and let me know if you have more feedback.

ldanzinger commented on ArcGISRuntimeSDKQt_CppSamples/Layers/QueryOGCAPICQLFilters/QueryOGCAPICQLFilters.cpp:
```diff
@@ -0,0 +1,111 @@
+using namespace Esri::ArcGISRuntime;
+
+QueryOGCAPICQLFilters::QueryOGCAPICQLFilters(QObject* parent /* = nullptr */):
+    QObject(parent),
```
Spacing for C++ code in the Qt SDK should be 2 space indentations instead of the 4 (that Qt Creator is set to by default). QML files stay at 4 space indentations

ldanzinger commented on ArcGISRuntimeSDKQt_CppSamples/Layers/QueryOGCAPICQLFilters/QueryOGCAPICQLFilters.qml:
```diff
@@ -0,0 +1,145 @@
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+import Esri.Samples 1.0
+import QtQuick.Controls 1.4
```
We should only be using QtQuick.Controls 2 in our samples. They have a different look and feel and are the more modern approach. In this case, both 1 and 2 are imported, but 2 is imported last, so any controls (like Button, ComboBox) will use version 1 instead of version 2

ldanzinger commented on ArcGISRuntimeSDKQt_CppSamples/Layers/QueryOGCAPICQLFilters/QueryOGCAPICQLFilters.cpp:
```diff
@@ -0,0 +1,111 @@
+  queryParams.setWhereClause(whereClause);
+  queryParams.setMaxFeatures(maxFeature.toUInt());
+
+  QDateTime fromDate = QDateTime::fromString(fromDateString,"mm-dd-yyyy");
+  QDateTime toDate = QDateTime::fromString(toDateString,"mm-dd-yyyy");
+  TimeExtent timeExtent(fromDate, toDate);
```
can be const

ldanzinger commented on ArcGISRuntimeSDKQt_CppSamples/Layers/QueryOGCAPICQLFilters/README.metadata.json:
```diff
@@ -0,0 +1,36 @@
+    "description": "Query data from an OGC API feature service using CQL filters.",
+    "ignore": false,
+    "images": [
+        "screenshot.png"
```
screenshot needs to be updated with the QtQuick Controls 2 update. Also, screenshots should be 500x500 pixels, and we generally try and take a platform agnostic screenshot (don't include the windows or mac top bar)

ldanzinger replied:
screenshot needs to be updated with the QtQuick Controls 2 update. Also, screenshots should be 500x500 pixels, and we generally try and take a platform agnostic screenshot (don't include the windows or mac top bar)

ldanzinger replied:
screenshot needs to be updated with the QtQuick Controls 2 update. Also, screenshots should be 500x500 pixels, and we generally try and take a platform agnostic screenshot (don't include the windows or mac top bar)

ldanzinger reviewed (approved):
Nice work @Gela. Screenshot needs updating, but other than that, this looks great.

tanneryould commented on ArcGISRuntimeSDKQt_QMLSamples/Layers/QueryOGCAPICQLFilters/QueryOGCAPICQLFilters.qml:
```diff
@@ -85,119 +97,118 @@ Rectangle {
+                                              "whereClause": whereClauseMenu.currentText,
+                                              "maxFeatures": maxFeatureField.text,
+                                              "timeExtent": time
+                                          })
```
```suggestion
                                          });
```

tanneryould commented on ArcGISRuntimeSDKQt_QMLSamples/Layers/QueryOGCAPICQLFilters/QueryOGCAPICQLFilters.qml:
```diff
@@ -85,119 +97,118 @@ Rectangle {
                     // Populate the feature collection table with features that match the parameters,
                     // clear the cache to prepare for the new query results, and store all table fields
-                    ogcFeatureCollectionTable.populateFromService(queryParams, true, []);
+                    ogcFeatureCollectionTable.populateFromService(queryParams,
+                                                                  true, [])
```
This seemed fine as it was. Any reason for the change? Also JS doesn't need them, but we include the semicolons regardless.
```suggestion
                    ogcFeatureCollectionTable.populateFromService(queryParams, true, []);
```

tanneryould commented on ArcGISRuntimeSDKQt_QMLSamples/Layers/QueryOGCAPICQLFilters/QueryOGCAPICQLFilters.qml:
```diff
@@ -45,7 +44,8 @@ Rectangle {
                         // populateFromService() will be called with the initial viewpoint extent
                         if (ogcFeatureCollectionTable.loadStatus === Enums.LoadStatusLoaded) {
-                            ogcFeatureCollectionTable.populateFromService(queryParameters, false, []);
+                            ogcFeatureCollectionTable.populateFromService(
+                                        queryParameters, false, [])
```
See similar suggestion below
```suggestion
                            ogcFeatureCollectionTable.populateFromService(queryParameters, false, []);
```

tanneryould commented on ArcGISRuntimeSDKQt_CppSamples/Layers/QueryOGCAPICQLFilters/QueryOGCAPICQLFilters.qml:
```diff
@@ -128,17 +125,19 @@ Item {
-                model.query(hereClauseMenu.currentText, maxFeatureField.text, fromDate.text, toDate.text);
+            onClicked: {
+                model.query(whereClauseMenu.currentText, maxFeatureField.text,
+                            fromDate.text, toDate.text)
```
```suggestion
                            fromDate.text, toDate.text);
```

tanneryould reviewed (approved):
Looks good to me, just a few insignificant suggested changes, but no need for re-review

AndrewBladon reviewed (approved):
The sample looks great @Gela!
